shared/bpm/bpm.h

```diff
@@ -0,0 +1,26 @@
+#ifndef BPM_H
+#define BPM_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* BPM callback. Allocation of BPM metrics data happens automatically
+ * with the first invokation of the callback associated with the output.
+ * Deallocation must be done explicitly with bpm_destroy(), after the
+ * callback is removed.
+ *
+ * BPM is designed to operate at the packet level. The bpm_injection()
+ * callback function must be registered and unregistered with
+ * obs_output_add_packet_callback() and obs_output_remove_packet_callback(),
+ * respectively.
+ */
+void bpm_injection(obs_output_t *output, struct encoder_packet *pkt,
```

Comment:
Microscopic nitpick: Should this perhaps be named `bpm_inject` to match the tense of `bpm_destroy`?

Reply:
I've renamed it to `bpm_inject()` just now. Also added a note about memset() to the documentation on `obs_output_add_packet_callback()`. 

Next step is squashing commits, which I'll start shortly.

Reply:
This feedback seems to have been addressed.